ginga/AstroImage.py

```diff
@@ -307,7 +276,8 @@ def update_metadata(self, key_dict):
             self.wcs.load_header(header)
 
     def has_primary_header(self):
-        return self._primary_hdr is not None
+        primary_hdr = self.metadata.get('primary_header', None)
+        return primary_hdr is not None
```

Comment:
Can this be simplified without actually trying to access the data that you never use?
```suggestion
        return 'primary_header' in self.metadata
```

Reply:
I think I'd like to keep the type check on the value of the primary_header key. At least checking that it is not `None`.  We could check that it is a subclass of `Header`...

Reply:
If you prefer to keep type check, I agree checking for `Header` might be more explicit. 😸